fabtests/multinode/src/core.c

```diff
@@ -460,18 +479,33 @@ int multinode_run_tests(int argc, char **argv)
 	if (ret)
 		return ret;
 
+	if (ft_check_opts(FT_OPT_PERF))
+		timers = malloc(sizeof(*timers) * opts.iterations * pm_job.num_ranks);
+
 	for (i = 0; i < NUM_TESTS && !ret; i++) {
 		printf("starting %s... ", patterns[i].name);
 		pattern = &patterns[i];
+
 		ret = multi_run_test();
-		if (ret)
+		if (ret) {
 			printf("failed\n");
-		else
-			printf("passed\n");
+			goto out;
+		}
+		printf("passed\n");
+
+		if (ft_check_opts(FT_OPT_PERF)) {
+			ret = multi_timer_analyze(timers, opts.iterations * pm_job.num_ranks);
+			if (ret)
+				goto out;
+			for (j = 0; j < opts.iterations * pm_job.num_ranks; j++)
+				free(timers[j]);
+		}
 
 		fflush(stdout);
 	}
-
+out:
+	for (j = 0; j < opts.iterations * pm_job.num_ranks; j++)
+		free(timers[j]);
```

Comment:
... you call free(timers[j]).  Which is it?  May be better to have timer_alloc/timer_free calls.